pkg/services/generic.go

```diff
@@ -333,3 +344,66 @@ func (s *sqlGenericService) treeWalkForSqlizer(listCtx *listContext, tslTree tsl
 
 	return tslTree, sqlizer, nil
 }
+
+func isJSONBSearch(search string) bool {
+	if strings.Contains(search, "->>") || strings.Contains(search, "@>") {
+		return true
+	}
+	return false
+}
+
+func validateJSONBSearch(search string) error {
+	// handle ->> search. for example: payload -> 'data' -> 'manifest' -> 'metadata' -> 'labels' ->> 'foo' = 'bar'
+	if strings.Contains(search, "->>") {
+		jsonbStr := stringSplitwithTrim(search, "->>")
+		if err := isValidFields(stringSplitwithTrim(jsonbStr[0], "->")); err != nil {
```

Comment:
do we ensure jsonbStr is always not nil

Reply:
if the search string has "->>", it can enter this method. that means jsonbStr[0] and jsonbStr[1] is always not nil. for example: "'labels'->>", the jsonbStr[0] is `labels` and jsonbStr[1] is empty.

Reply:
what if only "->>"? Also what if there are two "->>" in it? Do we allow only one?

Reply:
```
		{
			"name":   "invalid search without field name",
			"search": "->>",
			"error":  "the search field name is invalid",
		},
		{
			"name":   "invalid search with two ->> symbols",
			"search": "payload -> 'data' -> 'manifest' -> 'metadata' -> 'labels'->> 'foo' = 'bar' AND 'labels'->> 'foo'",
			"error":  "the search value is invalid",
		},

```
Added 2 tests to cover the above case. Yes. we allow only one `->>` symbol.